src/languages/dos.js

```diff
@@ -148,7 +148,7 @@ export default function(hljs) {
       {
         className: 'function',
         begin: LABEL.begin,
-        end: 'goto:eof',
+        end: '\n',
```

Comment:
Ok, that was a weird end condition, but now `D:` is incorrectly detected as a label, is it not?  From what I'm reading labels always start with `:` not end with it... so could we just update our label matcher to something like?

```
[start of line]:[label_name]
```

I think that would wrap this up nicely.

Reply:
That makes sense, ive added expressions like `[start of line]:[end of line]` to illegal so they wont be matched anymore.